src/http_workers/mod.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{storage::Storage, types::QueuedMessage};
+
+use self::worker_pool::WorkerPool;
+
+mod worker;
+mod worker_pool;
+
+type Job = Box<dyn FnOnce() + Send + 'static>;
```

Comment:
I understand you want to create a generic worker pool. But probably this can be an over-kill to this project. Let's try to avoid dynamic allocations in a high performance project like this. 

It's a nice side project idea to create a generic worker pool. May be start a new crate. but I will still advice against dynamic allocation if it's not needed. 

May be a better (yet static) allocation idea is 

```rust
trait Work {
  type Job: Send + 'static; 
  fn run(&self, job: Self::Job);
}

struct Worker<T: Work> {
   work: T,
  // hold Worker state like the channels sent to it from the pool
}
```
Then Work can implement this trait. This way Workers will have `function` statically dispatched not dynamically dispatched and also will have no data on heap. 

May be if you want you can implement this here directly like this, then later you can create a new crate based on the same idea.

Reply:
This way, a Job can become an `enum` itself. Worker once spawned and has a reference to the channel (as discussed below) we don't need to hold a reference to it anymore even.

Reply:
As an example of what I mean, i create this https://gist.github.com/muhamadazmy/50f7ca128fd700939838f66dc4e7c640 which shows how you can separate the worker (function) form the worker channels and pool implementation

Reply:
sorry, I am not make sure that I understand this comment. But, actually, I send the Job in a message (`http_workers::Msg(Job)`) as a closure to the worker through the `worker object`. so that I declared it in the heap and puth the  trait FnOnce().

Reply:
Let me ask you this. Why the Job is a Box ?